spec/spec_models.cr

```diff
@@ -400,6 +401,92 @@ require "uuid"
     timestamps
   end
 
+  class ManualColumnType < Granite::Base
+    adapter {{ adapter_literal }}
+    table_name manual_column_types
+
+    field foo : UUID, column_type: "FOO"
+  end
+
+  struct MyType
+    include JSON::Serializable
+
+    def initialize; end
+
+    property name : String = "Jim"
+    property age : Int32 = 12
+  end
+
+  enum MyEnum
+    Zero
+    One
+    Two
+    Three
+    Four
+  end
+
+  {% if env("CURRENT_ADAPTER") == "pg" %}
+    class ConverterModel < Granite::Base
+      adapter {{ adapter_literal }}
+      table_name converters
+
+      field binary_json : MyType, column_type: "BYTEA", converter: Granite::Converters::Json(MyType, Bytes)
+      field string_json : MyType, column_type: "JSON", converter: Granite::Converters::Json(MyType, JSON::Any)
+      field string_jsonb : MyType, column_type: "JSONB", converter: Granite::Converters::Json(MyType, JSON::Any)
+
+      field smallint_enum : MyEnum, column_type: "SMALLINT", converter: Granite::Converters::Enum(MyEnum, Int16)
+      field bigint_enum : MyEnum, column_type: "BIGINT", converter: Granite::Converters::Enum(MyEnum, Int64)
+      field string_enum : MyEnum, column_type: "TEXT", converter: Granite::Converters::Enum(MyEnum, String)
+      field enum_enum : MyEnum, column_type: "my_enum_type", converter: Granite::Converters::Enum(MyEnum, Bytes)
+      field binary_enum : MyEnum, column_type: "BYTEA", converter: Granite::Converters::Enum(MyEnum, Bytes)
+
+      field string_uuid : UUID, converter: Granite::Converters::Uuid(String) # Test PG native UUID type
+      field binary_uuid : UUID, column_type: "BYTEA", converter: Granite::Converters::Uuid(Bytes)
+
+      field numeric : Float64, column_type: "DECIMAL(21, 20)", converter: Granite::Converters::PgNumeric
+    end
+    ConverterModel.exec(<<-TYPE
+      DO $$
+      BEGIN
+          IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname = 'my_enum_type') THEN
+            CREATE TYPE my_enum_type AS ENUM ('Zero', 'One', 'Two', 'Three', 'Four');
+          END IF;
+      END$$;
+      TYPE
+    )
+  {% elsif env("CURRENT_ADAPTER") == "sqlite" %}
+    class ConverterModel < Granite::Base
+      adapter {{ adapter_literal }}
+      table_name converters
+
+      field binary_json : MyType, column_type: "BLOB", converter: Granite::Converters::Json(MyType, Bytes)
```

Comment:
This seems very verbose with a lot of types being defined:  MyType, BLOB, Json, MyType, Bytes.  Anyway to reduce complexity here?

Reply:
Not without manually defining a converter for every case. Like `PgJsonStringConverter`, `JsonStringConverter`, etc.

In reality the end user would most likely only have to do like:
`field binary_json : MyType, converter: Granite::Converters::Json(MyType, Bytes)` as the `column_type` is currently only used within Granite's migrator for specs.